### PR TITLE
Add GolfScript

### PR DESCRIPTION
**Pull request by MeWhenI**
Not gonna be winning any GolfScript golds at this stage, buts its a fully functional MVP. ⛳

### CODE REVIEW

**Comment by MichalMarsalek** [comment]
This needs main merged to work with the readonly IR and also some tests.

File: src/languages/golfscript/index.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  extension: "txt", // Golfscript doesn't appear to have a custom extension
+  emitter: emitProgram,
+  golfPlugins: [flipBinaryOps, evalStaticExpr, golfLastPrint()],
+  emitPlugins: [useIndexCalls()],
+  finalEmitPlugins: [
+    mapOps([
+      ["true", (_) => id("1", true)],
+      ["false", (_) => id("0", true)],
+      ["println", (x) => functionCall(x, "puts")],
+      ["print", (x) => functionCall(x, "print")],
```
Comment (jared-hughes) [inline src/languages/golfscript/index.ts:28]: Does the current system make it feasible to use `p` for printing integers?

File: src/languages/golfscript/index.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      ["neq", "=!"],
+      ["geq", "(>"],
+      ["join", "''*"],
+      ["text_reversed", "-1%"],
+      ["text_get_char", "=[]+''+"],
+      ["byte_to_char", "[]+''+"],
+      ["max", "[]++$1="],
+      ["min", "[]++$0="],
+
+      ["argv_get", "a\\="],
```
Comment (jared-hughes) [inline src/languages/golfscript/index.ts:78]: Is the swap `\` necessary? In testing, it seemed to work with no `\`.

**Review by jared-hughes — APPROVED** [review]

File: src/languages/golfscript/index.ts (hunk 2 above)
Comment (MeWhenI) [reply src/languages/golfscript/index.ts:78]: No, good point

File: src/languages/golfscript/index.ts (hunk 1 above)
Comment (MeWhenI) [reply src/languages/golfscript/index.ts:28]: It does, yes. In the current system, `println` and `print` can be feasibly implemented as "n" and "", though this likely isn't a very future-proofed approach.

File: src/languages/golfscript/index.ts (hunk 2 above)
Comment (MeWhenI) [reply src/languages/golfscript/index.ts]: Fixed

**Pull request merged**